src/libraries/System.Text.Json/gen/JsonSourceGenerator.Emitter.cs

```diff
@@ -610,13 +609,16 @@ private string GenerateForObject(TypeGenerationSpec typeMetadata)
                         sb.Append($@"
     {PropVarName}[{i}] = {JsonMetadataServicesTypeRef}.CreatePropertyInfo<{memberTypeCompilableName}>(
         options,
-        isProperty: {memberMetadata.IsProperty.ToString().ToLowerInvariant()},
+        isProperty: {ToCSharpKeyword(memberMetadata.IsProperty)},
```

Comment:
This is now at 14 parameters. I wonder if we should just create the JsonPropertyInfo in one call (with minimal parameters) and then set the individual values, or perhaps create a struct and pass by reference.

Another option is to pass a struct by ref. This could be made binary forward-compatible (e.g. we only add fields) but would need to be discussed\designed.

Reply:
Agree with the feedback that this is getting too long. I like the struct-by-ref idea and will bring it up in design review.